docs/tasks.md

```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
 [Creative Commons Attribution 4.0 License](https://creativecommons.org/licenses/by/4.0/),
 and code samples are licensed under the
 [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0).
+
+## Debugging a Task
+
+In software, we do things not because they are easy, but because we think they will be.
+Lots of things can go wrong when writing a Task.
+This section contains some tips on how to debug one.
+
+### Inspecting the Filesystem
+
+One common problem when writing Tasks is not understanding where files are on disk.
+For the most part, these all live somewhere under `/workspace`, but the exact layout can
+be tricky.
+To see where things are before your task runs, you can add a step like this:
+
+```yaml
+- name: build-and-push-1
+  image: ubuntu
+  command:
+  - /bin/bash
+  args:
+  - -c
+  - |
+    set -ex
+    find /workspace
+```
```

Comment:
oh yeah good call, i do this all the time too - to the point where maybe we should consider adding a "debug" mode that does something like this?